Add Ocean data view tutorial and new topic (#4405)
galaxyproject/training-material: topics/earth-system/tutorials/ocean-data-view/tutorial.md

>
> 9. On th top left click on **File** select **Exit**
> 10. If you want to save the other window also click on **Yes**. Here we don't need it so click **No**.
>
> You can now go back on your Galaxy instance.
{: .hands_on}

Now, after waiting for everything to turn green in your history, you can see 2 new outputs "1900392_prof_scatter" and "1900392_prof_map"

> <hands-on-title>View in Galaxy</hands-on-title>
> In the history panel click on the {% icon galaxy-eye %} (eye) icon of your output.
>
> You can now visualize the outputs in galaxy middle pannel. 
{: .hands_on}

![Image in the middle pannel of "1900392_prof_scatter"](../../images/coastal_water_dyn/scatter.png)
![Image in the middle pannel of "1900392_prof_map"](../../images/coastal_water_dyn/map.png)

# Conclusion

Great you now know how to use argo floater data to create profiles curves in Ocean Data View (ODV) within Galaxy.

# Extra information
Coming up soon follow ups tutorials on Coastal Water Dynamics workflow. Keep an {% icon galaxy-eye %} open if you are interested !
Theese tutorials are made within the [EOSC FAIR-EASE project](https://fairease.eu/).

> ![Project funded by the european union](../../images/coastal_water_dyn/eu_fe.png)

Reviewer: These should be recored as a 'funding' key in the `contributions:` E.g.:

```
contributions:
  authorship:
    - shiltemann
  editing:
    - bebatut
    - hexylena
  funding:
    - fairease
```

see the [full gtn tutorial schema documentation](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/schemas/tutorial.html) for details on the individual roles.

You'll need to add a contributor with the right name to CONTRIBUTORS.yaml

> * Tools on the left
> * Viewing panel in the middle
> * History of analysis and files on the right
>
> ![Screenshot of the Galaxy interface, the tools panel is on the left, the main panel is in the center, and the history is on the right.](../../images/coastal_water_dyn/galaxy_interface.png "Galaxy interface explanation")
>
> The first time you use Galaxy, there will be no files in your history panel.
{: .details}

> <hands-on-title>Deploy your own ODV instance</hands-on-title>
>
> 1. Create a new history for this tutorial nd give it a name (example: “ODV for Argo”) for you to find it again later if needed.
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
> 
> 2. Import netCDF data from amazonaws with this link an rename it "1900392_prof.nc" instead of "New File"
>
>    ```
>    https://argo-gdac-sandbox.s3-eu-west-3.amazonaws.com/pub/dac/coriolis/1900392/1900392_prof.nc
>    ```
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}  
>  
> 3. Select your new data before deploying {% tool [ODV instance](https://earth-system.usegalaxy.eu/root?tool_id=interactive_tool_odv) %} by pressing **execute**
> 4. Access ODV 
>

> > > <tip-title> Reminder on how to acces interactive tool </tip-title>
> > >  A URL will be displayed at the top of the Galaxy center panel. If you don't see it, you can see and access it through the "Active InteractiveTools" space of the "User" menu or you can click on {% icon galaxy-eye %} on the tool in the history.
> > {: .tip}

Reviewer: ```suggestion
> {% snippet faqs/galaxy/interactive_tools_open.md tool="ODV" %}
```

>
{: .hands_on}

> <details-title> Some complementary information about your data </details-title>

> > Argo is an international program to observe the interior of the ocean with a fleet of profiling floats drifting in the deep ocean currents (check it [here](https://argo.ucsd.edu)). Argo GDAC is a dataset of 5 billion in situ ocean observations from 18.000 profiling floats (4.000 active) which started 20 years ago. Argo GDAC dataset is a collection of 18.000 NetCDF files. It is a major asset for ocean and climate science, a contributor to IOCCP reports. For more informations click [here]( https://doi.org/10.17882/42182)

Reviewer: ```suggestion
> Argo is an international program to observe the interior of the ocean with a fleet of profiling floats drifting in the deep ocean currents (check it [out on the argo site](https://argo.ucsd.edu)). Argo GDAC is a dataset of 5 billion in situ ocean observations from 18.000 profiling floats (4.000 active) which started 20 years ago. Argo GDAC dataset is a collection of 18.000 NetCDF files. It is a major asset for ocean and climate science, a contributor to IOCCP reports. {% cite argodata %}
```

The 'here' link is an issue for people navigating with screenreaders, having a more descriptive link is helpful for them.

Separately we encourage that for citable objects, the DOI is added to the bibliography and referenced that way. You can add

```
@misc{argodata,
  doi = {10.17882/42182},
  url = {https://www.seanoe.org/data/00311/42182/},
  author = {{Argo}},
  keywords = {float,  Argo,  global ocean observing system,  ocean circulation,  in-situ,  ocean pressure,  sea water salinity,  sea water temperature,  multi-year,  weather climate and seasonal observation,  global-ocean},
  title = {Argo float data and metadata from Global Data Assembly Centre (Argo GDAC)},
  publisher = {SEANOE},
  year = {2023},
  copyright = {Creative Commons Attribution 4.0 International}
}
```

to the tutorial.bib file

>
> These data are available on AWS registry of Open Data. This registry exists to help people discover and share datasets that are available via AWS resources. [See recent additions and learn more about sharing data on AWS](https://registry.opendata.aws/).
{: .details}

# Ocean Data View
## Map your data

> <tip-title>Copy pasting between computer and ODV</tip-title>
> You can expand the ODV left panel (where there are 3 dots, vertically) to access the "clipboard" menu, and paste the content you want to paste on a ODV form. From there you can copy paste everything from one side to the other. Then, click outside of this panel to collapse it.
> 
> ![Image showing in transparent on the left of the ODV interface the clipboard](../../images/coastal_water_dyn/clipboard.png)
{: .tip}

> <tip-title>ODV - Disconnected</tip-title>

> > If at one point your ODV interface becomes grey with a red pannel "X ODV - Disconnected", do NOT panik ;) you just need to reload your tab (circular arrow top left)

Reviewer: ```suggestion
> If at one point your ODV interface becomes grey with a red panel "X ODV - Disconnected", do NOT panic ;) you just need to reload your tab (circular arrow top left)
```

{: .tip}

> <hands-on-title>Loading Argo data</hands-on-title>
>
> 1. Go to the top left of ODV and click on "Open..."
> 2. Click on "1900392_prof.nc" of the pop-up screen and on  **Open**
> 3. Press **Next >** for "Select Dimensions (Step 1 of 4)"
> 4. Press **Next >** for "Associate Meta Variables (Step 2 of 4)"
> 5. Press **Next >** for "Select Primary Variable (Step 3 of 4)"
> 6. Press **Finish** for "Subset Dimensions (Step 4 of 4)"
> 7. "Enter netCDF Setup Name:" write "argo_prof_data", then, press **OK**
{: .hands_on}
![Map of all the stations of the Argo glider](../../images/coastal_water_dyn/argo_stations.png)

> <question-title></question-title>
>
> 1. What are the longitude and latitude of the red dot ?
>
> > <solution-title></solution-title>
> >
> > 1. On the to right window you can read Longitude 7°E and Latitude 21°S.
> >
> {: .solution}
>
{: .question}

## Create profiles curves

> <hands-on-title>Create a multiple windows view</hands-on-title>
> 1. On the top left press **+**
> 2. Click on **2 Station Windows**

> > ![Image showing where to go to have the 2 STATION Windows](../../images/coastal_water_dyn/2_station.png)

Reviewer: ```suggestion
>    ![Image showing where to go to have the 2 STATION Windows](../../images/coastal_water_dyn/2_station.png)
```

you should indent images to match the start of the text, otherwise it will interrupt your numbered list, and you'll see `1. ... 2.... [image] 1...` which probably isn't what you wanted

> 3. Two more slots appear follow the indication and press + on your keyboard (or SHIFT and +).
>
> Here you have created a multiple view on ODV
{: .hands_on}

> <hands-on-title>Create profiles</hands-on-title>
> 1. On the biggest window click on the right side of your mousse
> 2. Select **X-Variable**, press **TEMP [degree_Celsius]** and OK.

> > ![Image of the selection of the pression](../../images/coastal_water_dyn/xvar_temp.png)
> > 3. Select **Y-Variable**, press **PRES [decibar]** and OK. Your profile is now uspide down. 
> > 4. To correct that click right again go on **Y-Variable** and on the bottom of the pop-up screen check the box "Reverse range" and OK.
> > ![ODV interface with the map and the 1st profile pression temperature](../../images/coastal_water_dyn/1st_profile.png)
> > 5. For the smallest window do the same but for the abscisse select **TEMP_ADJUSTED [degree_Celsius]**. This can be usefull to compare the 2 curves.
> > ![Resulting image of the 2 windows configurated](../../images/coastal_water_dyn/2_profiles.png)

Reviewer: ```suggestion
>    ![Image of the selection of the pression](../../images/coastal_water_dyn/xvar_temp.png)
> 3. Select **Y-Variable**, press **PRES [decibar]** and OK. Your profile is now uspide down. 
> 4. To correct that click right again go on **Y-Variable** and on the bottom of the pop-up screen check the box "Reverse range" and OK.
>    ![ODV interface with the map and the 1st profile pression temperature](../../images/coastal_water_dyn/1st_profile.png)
> 5. For the smallest window do the same but for the abscisse select **TEMP_ADJUSTED [degree_Celsius]**. This can be usefull to compare the 2 curves.
>    ![Resulting image of the 2 windows configurated](../../images/coastal_water_dyn/2_profiles.png)
```

>
> You now know how to create multiple profiles curves on ODV from netCDF data.
{: .hands_on}

> <question-title></question-title>
>

> > 2. How do you do if you want to compare the profiles with another station ?
> >
> > > <solution-title></solution-title>
> > >
> > > 2. With your keyboard you can use the arrows left and right and go through the different stations one-by-one.
> > > ![See the profiles of another station](../../images/coastal_water_dyn/2_curves.png)

Reviewer: ```suggestion
> How do you do if you want to compare the profiles with another station ?
>
> > <solution-title></solution-title>
> >
> > With your keyboard you can use the arrows left and right and go through the different stations one-by-one.
> > ![See the profiles of another station](../../images/coastal_water_dyn/2_curves.png)
```

with just one question probably not necessary to number

> >
> {: .solution}
>
{: .question}

> <hands-on-title>Change the station of reference</hands-on-title>
> 1. Go on the map on the bottom left 
> 2. Click right and select **Station Filter**
> ![Image on how to select the option Station Filter](../../images/coastal_water_dyn/station_filter.png)
> 3. Then, select and click on **Customize**
> 4. In "Station ID" write the number of the new station of reference (both boxes)
> 

> > Everything has dissapeared !!! Don't panik it's totally normal ;)
> > - Press **+** on your keyboard (or SHIFT and +)
> > 
> > The curves reapeared ! And they correspond to your knew station of reference !
> > 
> > To make the other stations reapear, to allow you to navigate in the other stations while conserving your new station of reference, go on the map once again.

Reviewer: ```suggestion
>    Everything has dissapeared !!! Don't panik it's totally normal ;)
>    - Press **+** on your keyboard (or SHIFT and +)
> 
>    The curves reapeared ! And they correspond to your knew station of reference !
> 
>    To make the other stations reapear, to allow you to navigate in the other stations while conserving your new station of reference, go on the map once again.
```
